contrib/mac/framework/Makefile

```diff
@@ -0,0 +1,187 @@
+	ln -s -f ../$(FRAMEWORK_NAME) $@
+$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).$(SOMAJOR).dylib: | $(framework_destdirs)
+	ln -s -f ../$(FRAMEWORK_NAME) $@
+$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).dylib: | $(framework_destdirs)
+	ln -s -f ../$(FRAMEWORK_NAME) $@
+$(DESTDIR)$(prefix)/$(framework_currver)/libexec: | $(framework_destdirs)
+	ln -s -f Helpers $@
+$(DESTDIR)$(prefix)/$(framework_currver)/share: | $(framework_destdirs)
+	ln -s -f Resources $@
+$(DESTDIR)$(prefix)/$(framework_currver)/include: | $(framework_destdirs)
+	ln -s -f Headers $@
+$(DESTDIR)$(prefix)/$(framework_currver)/etc: | $(framework_destdirs)
+	ln -s -f Resources $@
+
+hier_symlinks: \
+	$(DESTDIR)$(prefix)/$(framework_currver)/bin \
+	$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).$(SOMAJOR).$(SOMINOR).dylib \
+	$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).$(SOMAJOR).dylib \
+	$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).dylib \
+	$(DESTDIR)$(prefix)/$(framework_currver)/libexec \
+	$(DESTDIR)$(prefix)/$(framework_currver)/share \
+	$(DESTDIR)$(prefix)/$(framework_currver)/include \
+	$(DESTDIR)$(prefix)/$(framework_currver)/etc
+
+$(DESTDIR)$(prefix)/$(framework_infoplist): | $(framework_destdirs)
+	/usr/libexec/PlistBuddy -x -c "Clear dict" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleName string $(FRAMEWORK_NAME)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleExecutable string $(FRAMEWORK_NAME)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleIdentifier string org.julialang.julia.lib" $@
+	#TODO: reflect dev/rc#/commithash in CFBundleVersion
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleVersion string $(JULIA_MAJOR_VERSION).$(JULIA_MINOR_VERSION).$(JULIA_PATCH_VERSION)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleShortVersionString string $(JULIA_MAJOR_VERSION).$(JULIA_MINOR_VERSION).$(JULIA_PATCH_VERSION)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleSignature string ???" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundlePackageType string FMWK" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleInfoDictionaryVersion string 6.0" $@
+	/usr/libexec/PlistBuddy -x -c "Add :NSHumanReadableCopyright string \"Copyright © 2009-2017 Julia project contributors (https://github.com/JuliaLang/julia/contributors). See LICENSE.md.\"" $@
+
+# from top-level Makefile:
+define stringreplace
+	$(build_depsbindir)/stringreplace $$(strings -t x - $1 | grep '$2' | awk '{print $$1;}') '$3' 255 "$(call cygpath_w,$1)"
+endef
+
+
+framework: $(DESTDIR)$(prefix)/$(framework_infoplist) | $(framework_current_symlinks) $(framework_version_symlink) $(framework_destdirs) hier_symlinks
+ifneq ($(DARWIN_FRAMEWORK),1)
+	$(error Darwin framework is not enabled. Please set DARWIN_FRAMEWORK=1)
+endif
+
+	$(MAKE) -C $(BUILDROOT) $(JULIA_BUILD_MODE) $(build_depsbindir)/stringreplace $(BUILDROOT)/doc/_build/html/en/index.html
+
+	# Add julia exe
```

Comment:
There were enough differences between the install target and the framework target that it was easier to make a new target.  In `Make.inc`, the bin/lib/docdir etc. are set to their equivalent framework directories, but that's mainly so that the `_rel` vars are correctly recorded in `build_h.jl`.   Running make install would not generate a correct framework structure and the result of make install would need to be corrected in several ways to yield a proper framework.

Also, a previous review suggested moving the framework target out of the main Makefile and under `contrib/mac`, making integration with and modifications to the install target more difficult.